File: tests/parsers/samples/code/jl/complex.jl
module ComplexMetrics

export distance, root_mean_square, build_default

distance(a::Number, b::Number; scale::Float64 = 1.0) = abs(a - b) * scale

function root_mean_square(values::AbstractVector{<:Number})
    isempty(values) && return 0.0
    total = sum(value -> value ^ 2, values)
    sqrt(total / length(values))
end

build_default() = (scale = 0.82,)

end

File: tests/parsers/samples/code/rs/complex.rs
pub struct ComplexMetrics {
    scale: f64,
}

impl ComplexMetrics {
    pub fn new(scale: f64) -> Self {
        Self { scale }
    }

    pub fn distance(&self, a: i32, b: i32) -> f64 {
        ((a - b).abs()) as f64 * self.scale
    }

    pub fn root_mean_square(&self, values: &[f64]) -> f64 {
        if values.is_empty() {
            return 0.0;
        }
        let total: f64 = values.iter().map(|v| v * v).sum();
        (total / values.len() as f64).sqrt()
    }
}

pub fn build_default() -> ComplexMetrics {
    ComplexMetrics::new(0.83)
}
